service/worker/service.go

```diff
@@ -252,12 +259,55 @@ func (s *Service) startArchiver(base service.Service, pFactory persistencefactor
 	}
 }
 
-func (s *Service) ensureSystemDomainExists(publicClient workflowserviceclient.Interface) {
-	domainClient := client.NewDomainClient(publicClient, nil)
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
-	_, err := domainClient.Describe(ctx, common.SystemDomainName)
+func (s *Service) ensureSystemDomainExists(pFactory persistencefactory.Factory, clusterName string) {
+	metadataProxy, err := pFactory.NewMetadataManager(persistencefactory.MetadataV1V2)
+	if err != nil {
+		s.logger.Fatal("error creating metadataMgr proxy", tag.Error(err))
+	}
+	defer metadataProxy.Close()
+	_, err = metadataProxy.GetDomain(&persistence.GetDomainRequest{Name: common.SystemDomainName})
+	switch err.(type) {
+	case nil:
+		return
+	case *shared.EntityNotExistsError:
+		s.logger.Info("cadence-system domain does not exist, attempting to register domain")
+		s.registerSystemDomain(pFactory, clusterName)
+	default:
+		s.logger.Fatal("failed to verify if cadence system domain exists", tag.Error(err))
+	}
+}
+
+func (s *Service) registerSystemDomain(pFactory persistencefactory.Factory, clusterName string) {
+	metadataV2, err := pFactory.NewMetadataManager(persistencefactory.MetadataV2)
+	if err != nil {
+		s.logger.Fatal("error creating metadataV2Mgr", tag.Error(err))
+	}
+	defer metadataV2.Close()
+	_, err = metadataV2.CreateDomain(&persistence.CreateDomainRequest{
+		Info: &persistence.DomainInfo{
+			ID:          uuid.New(),
```

Comment:
maybe we should hard code the uuid?

Reply:
Went back & forth on this one - hardcoded it in the end